src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -872,7 +889,8 @@ public synchronized HttpResponse doStop() {
         if (e != null) {
             return e.doStop();
         } else {
-            return doKill();
+            doKill();
+            return HttpResponses.forwardToPreviousPage();
```

Comment:
Does the double response break anything?

Reply:
🤔  possibly a stack trace that a response has been committed.  I need to manually confirm :)